keps/sig-testing/20190118-breaking-apart-the-kubernetes-test-tarball.md

```diff
@@ -0,0 +1,274 @@
+but this repo is also marked deprecated and read-only, and there have been no
+changes since July 2018.
+
+In vmware/simple-k8s-test-env, the `sk8.sh` file uses
+[`kubernetes-test.tar.gz`](https://github.com/vmware/simple-k8s-test-env/blob/master/sk8.sh#L4481),
+and this repo seems actively maintained, so we should make sure this continues
+working.
+
+The [reference](https://github.com/knative/test-infra/blob/8ef3dc1c2ed07e64024bc68c9dbd1a2e10e9e975/scripts/e2e-tests.sh#L118-L120)
+to `kubernetes-test.tar.gz` in knative/test-infra is hilarious.
+
+### Risks and Mitigations
+
+It's hard to tell who uses these test tarballs outside the core project and
+without tools like `kubetest`. We'll need to broadcast this change widely
+so that any downstream users are aware of the incompatible changes.
+
+This is an inherently breaking change, however, so we must decide when to
+cause the break. Assuming this change is targeted for the 1.14 release:
+
+1. We can continue to produce a mondo-tarball for 3 releases, along with new
+   split tarballs; i.e., both 1.14 and 1.15 would contain both split and mondo
+   test tarballs, while 1.16 would only use a split test tarball. This way one
+   could continue to use the mondo tarball through the 1.15 release cycle,
+   and then switch to using split test tarballs for 1.16, as all supported
+   releases would then be producing split test tarballs.
+2. We can make a clean break for 1.14, not producing any mondo test tarballs.
+   Downstream users would need to account for the break immediately,
+   and would also need to special-case for older releases that still use the
+   mondo test tarball.
+3. A somewhat hybrid approach, mixing 1 and 2 backwards in time:
+   a. Produce both mondo test tarballs and split test tarballs on master for
+      a few weeks.
+   b. Backport split tarballs to older releases still supported (1.11 through
+      1.13), but continue to produce mondo test tarballs. We would never remove
+      the mondo test tarballs from these releases, instead continuing to
+      produce both.
+   c. Update all test infrastructure to use split test tarballs
+   d. Remove the mondo test tarball from 1.14 before the first beta release.
+
+Given the Kubernetes [deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/),
+we should probably go for option 1 and continue to distribute mondo and split
+test tarballs for the 1.14 release, and possibly for several releases
+thereafter. (It's not entirely clear exactly which deprecation policy applies
+to this change, however.)
+
+We'll mark the mondo test tarball as deprecated in the 1.14 release.
+
+### Test Plan
+
+We'll start by building both the mondo test tarball and split test tarballs in
+CI, and then update test infrastructure to use the new split tarballs. We will
+monitor TestGrid to ensure that nothing is noticeably broken by the change.
```

Comment:
which testgrid? I'm assuming something like "All jobs but the ones we really care about sig-release-master-blocking, sig-release-master-informing, sig-release-master-upgrades"